src/lib/badge/badge.ts

```diff
@@ -47,46 +37,62 @@ declare global {
  * @slot start - Content placed before the default content.
  * @slot end - Content placed after the default content.
  *
+ * @state dot - The badge is rendered as a dot.
+ * @state strong - The badge has a stronger visual appearance.
+ * @state hide - The badge is hidden.
+ *
  * @cssclass forge-badge - The badge class _(required)_.
  * @cssclass forge-badge--dot - Renders the badge as a dot.
  * @cssclass forge-badge__icon - Styles a child element as an icon.
  */
-@customElement({
-  name: BADGE_CONSTANTS.elementName
-})
-export class BadgeComponent extends BaseComponent implements IBadgeComponent {
+@customElement(BADGE_TAG_NAME)
+export class BadgeComponent extends LitElement implements IBadgeComponent {
+  public static styles = unsafeCSS(styles);
+
+  #internals: ElementInternals;
+
   constructor() {
     super();
-    attachShadowTemplate(this, template, styles);
+    this.#internals = this.attachInternals();
   }
 
-  public get dot(): boolean {
-    return this.hasAttribute(BADGE_CONSTANTS.attributes.DOT);
-  }
-  public set dot(value: boolean) {
-    this.toggleAttribute(BADGE_CONSTANTS.attributes.DOT, value);
-  }
+  /** Controls whether the badge will be a small dot without any content visible. */
+  @property({ type: Boolean, reflect: true })
+  public dot = false;
 
-  public get theme(): BadgeTheme {
-    return (this.getAttribute(BADGE_CONSTANTS.attributes.THEME) as BadgeTheme) ?? BADGE_CONSTANTS.defaults.THEME;
-  }
-  public set theme(value: BadgeTheme) {
-    this.setAttribute(BADGE_CONSTANTS.attributes.THEME, value);
-  }
+  /** The theme of the badge. */
+  @property({ type: String, reflect: true })
+  public theme: BadgeTheme = 'default';
```

Comment:
I chose not to for this first pass, but should we consider also using CSS custom states for these common `theme` properties across components? What sort of naming convention would we use? `:state(theme-primary)` or `:state(primary)`?

Reply:
I don't think so. I've been reserving custom states for particular things like `disabled`, `open`, etc. It feels cleaner for them to refer to behavioral changes in the components instead of purely stylistic one. We can always add states later if there's a need, anyway 🤷